### DISCUSSION TITLE
A single quote yields triple quotes in YAML output

### DISCUSSION THREAD

**Discussion by antonalekseev** [TomWright/dasel#200] [Q&A] [answered]
For example, if I want to use an environment variable as a part of the value. The desirable result would be `foo: '/usr/local/bin/zsh'`

```
❯ echo $SHELL
/usr/local/bin/zsh
❯ echo ""|dasel put string -p yaml 'foo' "'$SHELL'"
foo: '''/usr/local/bin/zsh'''
```

Escaping doesn't help

```
❯ echo ""|dasel put string -p yaml 'foo' "\'$SHELL\'"
foo: \'/usr/local/bin/zsh\'
```

Using just single quotes predictably prevents variable expansion

```
❯ echo ""|dasel put string -p yaml 'foo' '$SHELL'
foo: $SHELL
```

```
❯ dasel --version
dasel version 1.22.1
```

**Comment by TomWright** [comment, marked as answer]
This should do the job:

```
echo ""|dasel put string -p yaml 'foo' "$SHELL"
```

There final argument will be the string value used. Using the single quotes is redundant and is being considered part of the value itself

**Reply by antonalekseev** [reply]
Hey Tom, thanks both for the swift response and the great tool we all now have in our toolboxes.

Actually, the aforementioned example was a simplified version of a problem that bothers me.

```
❯ export JOB_NAME=foo ; echo"" | dasel put object -p yaml -t string -t string "\.\.$JOB_NAME\.rules.rules.[]" if="$JOB_NAME != \"$JOB_NAME\" && \$CI_PIPELINE_SOURCE == \"trigger\"" when="never"
.foo.rules:
  rules:
  - if: foo != "foo" && $CI_PIPELINE_SOURCE == "trigger"
    when: never
```
In GitLab CI docs, strings that contain special characters are usually enclosed in a single quotes (`'foo != "foo" && $CI_PIPELINE_SOURCE == "trigger"'`, but according to the YAML spec only strings that _start_ with a special character should be. So `dasel` probably knows better when to quote:

```
❯ export JOB_NAME=foo ; echo "" | dasel put string -p yaml 'name' "&$JOB_NAME"
name: '&foo'
```
